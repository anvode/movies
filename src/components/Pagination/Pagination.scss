.pagination {
    &__btn {
        box-shadow: none;
        margin: 0 0.2rem 0 0;
        padding: 0.5rem;
        text-decoration: none;

        &.is--current {
            background: var(--color-base);
            color: var(--color-white);
            cursor: default;
        }
    }
}
