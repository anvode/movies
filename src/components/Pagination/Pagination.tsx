import * as React from 'react';
import { useSelector, useDispatch } from 'react-redux';

import { RootState } from '../../redux/root-reducer';
import { fetchMovies } from '../../redux/movies/movies.utils';

import './Pagination.scss';

export interface PaginationProps {}

const Pagination: React.FC<PaginationProps> = (props) => {
    const { currentPage, pages, searchQuery } = useSelector((state: RootState) => state.movies);
    const paginationList = Array.from(new Array(pages),(_,index)=> index + 1);
    const dispatch = useDispatch();

    const handleClick = (id: number) => (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
        event.preventDefault();

        if (id === currentPage) return;

        dispatch(fetchMovies(searchQuery, id));
    };

    return <div className="main">
        <div className="container">
            {paginationList.map(page => {
                const current = page === currentPage ? ' is--current' : '';
                return <button onClick={handleClick(page)} key={page} className={`pagination__btn btn${current}`}>{page}</button>;
            })}
        </div>
    </div>;
};

export default Pagination;