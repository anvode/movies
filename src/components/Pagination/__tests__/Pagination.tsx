import React from 'react';
import Store from '../../../redux/store';
import { render } from '@testing-library/react';
import Pagination from '../Pagination';
import {Provider} from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

describe('Pagination', () => {
    it('renders without crashing', () => {
        const { asFragment } = render( <Provider store={Store}><BrowserRouter><Pagination /></BrowserRouter></Provider>);
        expect(asFragment()).toMatchSnapshot();
    });
});