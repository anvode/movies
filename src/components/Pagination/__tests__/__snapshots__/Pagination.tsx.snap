// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="container"
    >
      <button
        class="pagination__btn btn is--current"
      >
        1
      </button>
    </div>
  </div>
</DocumentFragment>
`;
